api/engine.py

```diff
@@ -363,14 +452,66 @@ def __get_logs():
     Returns:
         an array with JSON events
     """
+    session = create_connection(__language())
     __api_key_check(app, flask_request, __language())
     try:
         host = __get_value(flask_request, "host")
     except Exception:
         host = ""
-    return jsonify(__logs_to_report_json(host, __language())), 200
+    data = __logs_to_report_json(host, __language())
+    json_object = json.dumps(data)
+    host_temp = session.query(HostsLog).filter(HostsLog.host==host).all()
+    dateFromDB = host_temp[0].date
+    dateFormat=datetime.strptime(dateFromDB, "%Y-%m-%d %H:%M:%S")
+    dateFormat = str(dateFormat).replace("-","_").replace(":","_").replace(" ", "_")
+    filename = "report-" + dateFormat+"".join(random.choice(string.ascii_lowercase) for x in range(10))
+    return Response(json_object,
+        mimetype='application/json',
+        headers={'Content-Disposition':'attachment;filename='+filename+'.json'})
 
 
+
+@app.route("/logs/get_csv", methods=["GET"])
+def __get_logs_csv():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        host = __get_value(flask_request, "host")
+    except:
+        host = ""
+    data = __logs_to_report_json(host, __language())
+    host_temp = session.query(HostsLog).filter(HostsLog.host==host).all()
+    dateFromDB = host_temp[0].date
+    dateFormat=datetime.strptime(dateFromDB, "%Y-%m-%d %H:%M:%S")
```

Comment:
@aman566  here the database query makes no sense because there could be hundreds of records with the same host. Using the date from the database only makes sense on a Report / specific ID. For hosts please revert back to the current date using now() and remove the database query here, because it slows down the export significantly (e.g. query finds all dates when uber.com was ever scanned and you are using just the first date).

Example: there is no date on a host, so nothing to mentally link this with:
![Screen Shot 2021-03-21 at 12 20 49](https://user-images.githubusercontent.com/20112179/111904680-0452b000-8a40-11eb-8fbe-82c5e80342e8.png)